libraries/AdaptiveExpressions/ExpressionFunctions.cs

```diff
@@ -2954,15 +2954,15 @@ private static IEnumerable<object> Flatten(IEnumerable<object> list, int dept)
                         object result = null;
                         if (args.Count == 2 && !args[1].IsInteger())
                         {
-                            error = $"The second {args[1]} parameter must be an integer.";
+                            error = $"The second parameter ${args[1]} must be an integer.";
                         }
 
                         if (error == null)
                         {
                             var digits = args.Count == 2 ? Convert.ToInt32(args[1]) : 0;
                             if (digits < 0 || digits > 15)
                             {
-                                error = $"The second parameter {args[1]} must be an integer between 0 and 15;";
+                                error = $"The second parameter {args[1]} must be an integer between 0 and 15.";
```

Comment:
This doesn't quite make sense. 15 digits looks like this:
123456789012345

However the error message is "between 0 and 15", which would only include the number 1-14.

Reply:
Also, this could throw, if the string is not a number. Consider "int32.TryParse()" and an error message on failure.

Reply:
15 digits means there are 15 fractional digits. Such as 21.123456789012345
For round function, the second paramter must be a 32-bit signed integer. However if Convert.ToInt32 failed, ApplyWithError method will catch the error message.
I think one thing we need to do is to warp all Convert.ToInt32 method to a new function. It can help us provide more detailed error message based on the context. I will create a issue on this.